spec/parallel_spec.rb

```diff
@@ -194,7 +194,7 @@ def cpus
     end
 
     it "raises when a thread raises" do
-      lambda{ Parallel.in_threads(2){|i| raise "TEST"} }.should raise_error("TEST")
+      lambda{ Parallel.in_threads(2){|i| raise "TEST" if i == 1} }.should raise_error("TEST")
```

Comment:
I added this because ```abort_on_exception = true``` was causing problems when two exceptions were thrown at the exact same time. Sometimes only one was getting rescued. I'm not sure if this is a bug in ruby threads or if it's a synchronization issue. But ether way, now that I think about it, we should account for that.

I might have found a better alternative to abort_on_exception so I'm going to play with that in the update to the PR